### PR TITLE
feat(msteams): handle installation expiration and uninstallation

### PR DESCRIPTION
**Pull request by scefali**
This PR adds handling for uninstallation and adds a 24-hour expiration for the welcome message we send during installation. Uninstallation happens via webhook which will delete the installation for ALL organizations which have it installed for that team. Generally, this would only be one organization but technically someone could use the welcome message link and set up the integration for multiple Sentry orgs. In either case, an integration can only be installed once per team so when it has been uninstalled in Teams, our access token stops working. So the integration would not be functional for any org. Note that you can still uninstall the integration in Sentry which just detaches your organization from the installation.

### CODE REVIEW

File: src/sentry/integrations/msteams/integration.py [hunk 1]
```diff
@@ -77,18 +79,17 @@ def get_pipeline_views(self):
...
         team_id = data["team_id"]
+        team_name = data["team_name"]
         service_url = data["service_url"]
 
+        # TODO: add try/except for request errors
         token_data = get_token_data()
-        access_token = token_data["access_token"]
 
-        client = MsTeamsPreInstallClient(access_token, service_url)
-        team = client.get_team_info(team_id)
```
Comment (scefali) [inline src/sentry/integrations/msteams/integration.py:86]: Realized we actually get the team name in the initial webhook so no need to make this extra call once we add it to state

File: src/sentry/integrations/msteams/webhook.py [hunk 2]
```diff
@@ -89,25 +94,87 @@ def post(self, request):
...
+            "team_id": team["id"],
+            "team_name": team["name"],
+            "service_url": data["serviceUrl"],
+            "expiration_time": int(time.time()) + INSTALL_EXPIRATION_TIME,
+        }
 
-                # sign the params so this can't be forged
-                signed_params = sign(**signed_data)
+        # sign the params so this can't be forged
+        signed_params = sign(**signed_data)
```
Comment (NisanthanNanthakumar) [inline src/sentry/integrations/msteams/webhook.py:134]: Is this creating a JWT?

Comment (NisanthanNanthakumar) [reply src/sentry/integrations/msteams/webhook.py:134]: ah nvm. I looked at the tests.

**Review by NisanthanNanthakumar — APPROVED** [review]

File: src/sentry/integrations/msteams/webhook.py (hunk 2 above)
Comment (scefali) [reply src/sentry/integrations/msteams/webhook.py:134]: @NisanthanNanthakumar Good question. It's not using JWT but I'm not actually sure what the name of this encryption is called from the docs: https://docs.djangoproject.com/en/1.10/topics/signing/

Comment (NisanthanNanthakumar) [reply src/sentry/integrations/msteams/webhook.py:134]: @scefali interesting! but we are using `@patch("jwt.decode")` in the tests?

Comment (scefali) [reply src/sentry/integrations/msteams/webhook.py:134]: @NisanthanNanthakumar The JWT token that's involved is in the header of the request that we verify. This signed parameter is something we generate for the link we send to users. Two different things going on here :)

File: src/sentry/integrations/msteams/webhook.py [hunk 3]
```diff
@@ -89,25 +94,87 @@ def post(self, request):
...
+            create_audit_entry(
+                request=request,
+                organization=org,
+                target_object=integration.id,
+                event=AuditLogEntryEvent.INTEGRATION_REMOVE,
+                actor_label="Teams User",
+                data={
+                    "provider": integration.provider,
+                    "name": integration.name,
+                    "team_id": team_id,
```
Comment (MeredithAnya) [inline src/sentry/integrations/msteams/webhook.py:175]: where does the `team_id` get used?

Comment (scefali) [reply src/sentry/integrations/msteams/webhook.py:175]: @MeredithAnya It's just another piece of data we can use if we need to debug something. It just gets stored in the JSON of the row.

Comment (MeredithAnya) [reply src/sentry/integrations/msteams/webhook.py:175]: okays, def have thought of the audit logs as primarily information for the user, not necessarily a debugging tool for us (in terms of storing extra data that doesn't get rendered), so wanted to make sure you weren't relying on it being shown

Comment (scefali) [reply src/sentry/integrations/msteams/webhook.py:175]: @scefali I see. I don't think any of the information in `data` is actually shown to the user unless I am mistaken: 
Screen Shot 2020-07-20 at 10 53 43 AM

**Pull request merged**